mozilla/addons-server
# Update task that resigns add-ons without a COSE signature

### Context

This is the main task we are going to use to resign add-ons without a COSE signature, to prepare for SHA-1 deprecation.

When ran, it will:
- Pick up publicly available add-ons for which the latest listed version (`current_version` internally) was created before the threshold date, then for each of them:
  - Clone this version into an entirely new one with a new version number considered higher than the existing one
  - Approve and sign that cloned version, replacing the existing one as the latest public listed version for each add-on
  - Notify affected developers

### Testing

The unit tests should cover a bunch of different scenarios. If you want to test it manually:

- Create some add-ons and versions depending on what you want to test. The script should only care about the `current_version` of public, non-invisible add-ons (`disabled_by_user=False`, `status=amo.STATUS_APPROVED`), if that version `created` date is from before 2019-04-05 and the add-on is not a dictionary.
- Run `manage.py --task=bump_and_resign_addons` (this will return and leave the celery workers to do the work when the tasks have been created, check your celery logs to see when it's finished)
- For add-ons matching the criteria described above, they should get a new `current_version` replacing the previous one. It should have a new version number higher than the one it's replacing (and with a `resigned1` suffix to distinguish from version numbers used by the developers), including in the manifest, and be signed.
- If there are developers/owners attached to the affected add-ons they should receive an email (locally, a `FakeEmail`).

Fixes #21996
